com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/modelinference/InferredModelAssociator.java

```diff
@@ -56,15 +60,20 @@
    * An adapter that holds the mapping between source- and inferred-model elements.
    */
   public static class Adapter extends AdapterImpl {
-    private final ListMultimap<EObject, EObject> sourceToInferredModelMap = LinkedListMultimap.create();
+    private final Map<EObject, Set<EObject>> sourceToTargetMap = Maps2.newLinkedHashMapWithExpectedSize(40);
+    private final Map<EObject, Set<EObject>> targetToSourceMap = Maps2.newLinkedHashMapWithExpectedSize(40);

```

Comment:
hmm, i wouldn't introduce a dependency just for Maps2 (espectially restricted to a class). I think the same result you can achieve by using Multimap.newSetMultimap and providing a map of some size and a supplier that would create instances of your linked list based set

Reply:
Multimap.newSetMultimap() would not work, as we in at least one of the maps need the elements to be ordered. It would have to be something like a LinkedHashMultimap.

But I would actually prefer to stick with LinkedHashMap for the following reason: In commit https://github.com/eclipse/xtext-extras/commit/7877735befe6b97a6decf068d9ce2641c42fe4e2 the implementation of JvmModelAssociator was tweaked for performance and I believe also memory overhad. While I think it is probably possible to improve this even more, I think copying this implementation is a good starting point.